libs/blocks/merch-card/merch-card.js

```diff
@@ -104,10 +104,11 @@ function addMerchCardGridsIfMissing(section) {
 
 const decorateMerchCardLinkAnalytics = (el) => {
   [...el.querySelectorAll('a')].forEach((link, index) => {
+    const config = getConfig();
     const heading = el.querySelector('h3');
-    const linkText = `${processTrackingLabels(link.textContent)}-${index + 1}`;
-    const headingText = heading ? `${heading.textContent}` : '';
-    const analyticsString = heading ? `${linkText}|${headingText}` : linkText;
+    const linkText = `${processTrackingLabels(link.textContent, config)}-${index + 1}`;
+    const headingText = heading ? `${processTrackingLabels(heading.textContent, config)}` : '';
+    const analyticsString = heading ? `${linkText}--${headingText}` : linkText;
```

Comment:
Judging by the unit test, it seems the `--` is intentional? I think one of the purposes of the recent analytics PR was to remove some special characters, as well as limiting the final string length. With this, you'll be defining something like `Link text-1--Heading text`. But if you apply the `processTrackingLabels` method to it, you'd be left with just `Link text 1 Heading `. Notice the string length is limited to 20 characters and the special characters are removed. Are you certain this is what your analysts want? Or should the `processTrackingLabels` be used for consistency?

Reply:
@overmyheadandbody Great questions, but to clarify, this PR is not about what analytics format we should use.  That has been debated and agreed to and already merged yesterday.  This is just about getting 2 blocks to conform to the new system.  That being said, I'm more than happy to answer your questions.

This is the link to the full documentation: https://wiki.corp.adobe.com/pages/viewpage.action?spaceKey=marketingtech&title=Analytics+in+Milo

And this is a 1 line summary of the syntax:
link name(20)-#--Last header before link(20)|s#|b#|blockname(15)|pzn experience name or "nopzn"(15)|pzn manifest filename or "nopzn"(20)

The text before the first | character is on the link itself: link name(20)-#--Last header before link(20)

processTrackingLabels won't remove the dashes because the link text and header are sent through separately, and the 2 outputs are then assembled with the proper dashes.  But yes, that function cleans the string and limits to 20 characters to enforce consistency. Also, I have another PR that will soon update the function to also check for EN equivalents of the string. https://github.com/adobecom/milo/pull/1609

The removal of special characters is something the gnav was already doing to their analytics and it reduces the chance of sending characters that AA can't process.